joplin/base/migrations/0033_auto_20180927_0500.py

```diff
@@ -0,0 +1,37 @@
+# Generated by Django 2.0.8 on 2018-09-27 05:00
+
+from django.db import migrations
+from pprint import pprint
+from wagtail.core.blocks.stream_block import StreamValue
+
+def better_steps(apps, schema_editor):
+    '''
+    We can't import the models directly as they may be a newer
+    version than this migration expects. We use the historical versions.
+    '''
+    ServicePage = apps.get_model('base', 'ServicePage')
+    ServicePageStep = apps.get_model('base', 'ServicePageStep')
+
+    for page in ServicePage.objects.all():
+      pageSteps = []
+      for step in ServicePageStep.objects.all():
+        if step.page_id == page.id:
+          streamStep = {
+            'type': 'basic_step',
+            'value': step.step_description
+          }
+          pageSteps.append(streamStep)
+      stream_block = page.steps.stream_block
+      page.steps = StreamValue(stream_block, pageSteps, is_lazy=True)
```

Comment:
This part is kind of weird, I followed the pattern I found [here](https://github.com/wagtail/wagtail/issues/2110#issuecomment-187821699), I'll do my best to explain it:

A streamfield has a StreamValue, which needs:
* stream_block - Still don't fully understand what it is, but since we have already initialized the StreamField on the ServicePage, we can just copy it back when we rebuild the steps.
* stream_data - In our case this is an array of steps, hence using pageSteps here.
* is_lazy - Again, not completely sure what this does, but it was already set to True in all of our steps StreamFields , so I'm making sure it stays that way.